amethyst_tiles/src/pass.rs

```diff
@@ -0,0 +1,503 @@
+                                    return None;
+                                }
+
+                                let sprite = &sprite_sheet.sprites[sprite_number];
+
+                                let transform =
+                                    convert::<_, Matrix4<f32>>(world_transform.matrix());
+                                let dir_x = transform.column(0) * sprite.width;
+                                let dir_y = transform.column(1) * -sprite.height;
+                                let pos = transform
+                                    * Vector4::new(
+                                        -sprite.offsets[0],
+                                        -sprite.offsets[1],
+                                        0.0,
+                                        1.0,
+                                    );
+
+                                let color = tile.tint(resources);
+                                let (r, g, b, a) = color.into_components();
+
+                                Some((
+                                    SpriteArgs {
+                                        dir_x: dir_x.xy().into_pod(),
+                                        dir_y: dir_y.xy().into_pod(),
+                                        pos: pos.xy().into_pod(),
+                                        u_offset: [sprite.tex_coords.left, sprite.tex_coords.right]
+                                            .into(),
+                                        v_offset: [sprite.tex_coords.top, sprite.tex_coords.bottom]
+                                            .into(),
+                                        depth: pos.z,
+                                        tint: [r, g, b, a].into(),
+                                    },
+                                    &sprite_sheet.texture,
+                                ))
+                            }?;
+
+                            let (tex_id, _) = textures_ref.insert(
+                                factory,
+                                resources,
+                                texture,
+                                hal::image::Layout::ShaderReadOnlyOptimal,
+                            )?;
+
+                            return Some((tex_id, batch_data));
+                        }
+                        return None;
+                    }
+                    None
+                })
+                .for_each_group(|tex_id, batch_data| {
+                    sprites_ref.insert(tex_id, batch_data.drain(..))
+                });
+        }
+
+        self.textures.maintain(factory, resources);
+
+        {
+            #[cfg(feature = "profiler")]
+            profile_scope!("write");
+
+            sprites_ref.prune();
+            self.vertex.write(
+                factory,
+                index,
+                self.sprites.count() as u64,
+                self.sprites.data(),
+            );
+        }
+
+        PrepareResult::DrawRecord
```

Comment:
Once ordered batcher is used, it can be used to detect changes and avoid recording when there weren't any.